sdk/storage/Azure.Storage.Blobs/src/BlobUriBuilder.cs

```diff
@@ -161,6 +161,20 @@ public string Query
         }
         private string _query;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BlobUriBuilder"/>
+        /// class for the specified Azure Storage Account.
+        /// </summary>
+        /// <param name="accountName">
+        /// The account name of the target storage account.
+        /// </param>
+        /// <remarks>
+        /// This defaults to Azure Public Cloud - i.e. a blob.core.windows.net suffix on the URL to the storage account
+        /// </remarks>
+        public BlobUriBuilder(string accountName) : this(new Uri($"https://{accountName}.blob.core.windows.net"))
```

Comment:
Accepting only the storage account name as a default, and assuming the ending suffix (e.g. blob.core.windows.net) is not a good idea.

We should be taking in the ending suffix of the endpoint. This would not be a good experience for customers who are new to storage and have a different type of endpoint details.